roofit/roofit/src/RooLagrangianMorphFunc.cxx

```diff
@@ -0,0 +1,3511 @@
+    obs = (RooRealVar *)(dynamic_cast<RooParamHistFunc *>(inputExample)
+                             ->paramList()
+                             .first());
+    obsExists = true;
+    this->_observables.add(*obs);
+  }
+
+  //   Note: "found!" will be printed if s2 is a substring of s1, both s1 and s2
+  //   are of type std::string. s1.find(s2)
+  // obtain the observable
+  if (!obsExists) {
+    if (mode && mode->InheritsFrom(TH1::Class())) {
+      //  cxcoutP(ObjectHandling) << "getObservable: creating new multi-bin
+      //  observable object " << obsname << std::endl;
+      TH1 *hist = (TH1 *)(inputExample);
+      obs = new RooRealVar(obsname, obsname, hist->GetXaxis()->GetXmin(),
+                           hist->GetXaxis()->GetXmax());
+      obs->setBins(hist->GetNbinsX());
+    } else {
+      //  cxcoutP(ObjectHandling) << "getObservable: creating new single-bin
+      //  observable object " << obsname << std::endl;
+      obs = new RooRealVar(obsname, obsname, 0, 1);
+      obs->setBins(1);
+    }
+    this->_observables.add(*obs);
+  } else {
+    // cxcoutP(ObjectHandling) << "getobservable: recycling existing observable
+    // object " << this->_observables.at(0) << std::endl;
+    if (strcmp(obsname, obs->GetName()) != 0) {
+      //  coutW(ObjectHandling) << " name of existing observable " <<
+      //  this->_observables.at(0)->GetName() << " does not match expected name
+      //  " << obsname << std::endl ;
+    }
+  }
+
+  TString sbw = Form("binWidth_%s", makeValidName(obs->GetName()).Data());
+  RooRealVar *binWidth = new RooRealVar(sbw.Data(), sbw.Data(), 1.);
+  double bw = obs->numBins() / (obs->getMax() - obs->getMin());
+  binWidth->setVal(bw);
+  binWidth->setConstant(true);
+  this->_binWidths.add(*binWidth);
+
+  return obs;
+}
+
+//#ifndef USE_MULTIPRECISION_LC
+//#pragma GCC diagnostic push
+//#pragma GCC diagnostic ignored "-Wunused-parameter"
+//#endif
+
+////////////////////////////////////////////////////////////////////////////////
+/// update sample weight (-?-)
+
+inline void RooLagrangianMorphFunc::updateSampleWeights() {
+  //#ifdef USE_MULTIPRECISION_LC
+  int sampleidx = 0;
+  auto cache = this->getCache(_curNormSet);
+  const size_t n(size(cache->_inverse));
+  for (auto sampleit : this->_config.getParamCards()) {
+    const std::string sample(sampleit.first);
+    // build the formula with the correct normalization
+    LinearCombination *sampleformula = dynamic_cast<LinearCombination *>(
+        this->getSampleWeight(sample.c_str()));
+    if (!sampleformula) {
+      // coutE(ObjectHandling) << Form("unable to access formula for sample
+      // '%s'!",sample.c_str()) << std::endl;
+      return;
+    }
+    cxcoutP(ObjectHandling)
+        << "updating formula for sample '" << sample << "'" << std::endl;
+    for (size_t formulaidx = 0; formulaidx < n; ++formulaidx) {
+      const SuperFloat val(cache->_inverse(formulaidx, sampleidx));
+      if (val != val) {
```

Comment:
```suggestion
#ifdef USE_UBLAS
      if (val != val) {
#else
      if (std::isnan(val)) {
#endif
```
I know I was commenting on this before, but I have a new Idea. We can use `std::isnan` for `double` and keep the current `!=` check for UBLAS, using preprocessor macros.